server/utils/db/db_utils.py

```
@@ -3021,6 +3006,51 @@ def initialize_tables():
+                        SELECT EXISTS (
+                            SELECT 1 FROM knowledge_base_memory WHERE content IS NOT NULL AND content != ''
+                            LIMIT 1
+                        )
+                    """)
+                    has_unmigrated_data = cursor.fetchone()[0]
+
+                    if has_unmigrated_data:
+                        from services.memory.migration_task import migrate_kb_to_memory
+                        migrate_kb_to_memory.delay()
+                        logging.info("Triggered automatic KB → memory migration task")
+            except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)